cmd/target_profile.go

```diff
@@ -0,0 +1,117 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+)
+
+type TargetProfileType int
+
+const (
+	TargetProfileTypeUnset = iota
+	TargetProfileTypeConnection
+)
+
+type TargetProfileConnectionType int
+
+const (
+	TargetProfileConnectionTypeUnset = iota
+	TargetProfileConnectionTypeSpanner
+)
+
+type TargetProfileConnectionSpanner struct {
+	endpoint string // Same as SPANNER_API_ENDPOINT environment variable
+	project  string // Same as GCLOUD_PROJECT environment variable
+	instance string
+	dbname   string
+	dialect  string
+}
+
+type TargetProfileConnection struct {
+	ty TargetProfileConnectionType
+	sp TargetProfileConnectionSpanner
+}
+
+type TargetProfile struct {
+	ty   TargetProfileType
+	conn TargetProfileConnection
+}
+
+func (trg TargetProfile) ToLegacyTargetDb() string {
+	switch trg.ty {
+	case TargetProfileTypeConnection:
+		{
+			conn := trg.conn
+			switch conn.ty {
+			case TargetProfileConnectionTypeSpanner:
+				{
+					sp := conn.sp
+					if len(sp.dialect) > 0 && strings.ToLower(sp.dialect) == "postgresql" {
+						return "experimental_postgres"
+					}
+					return "spanner"
+				}
+			default:
+				return "spanner"
+			}
+		}
+	default:
+		return "spanner"
+	}
+}
+
+// Target profile is passed as a list of key value pairs on the command line.
+// Today we support only direct connection as a valid target profile type, but
+// in future we can support writing to CSV or AVRO as valid targets.
+//
+// Among direct connection targets, today we only support Spanner database.
+// TargetProfileConnectionType can be extended to add more databases.
+// Users can specify the database dialect, instance, database name etc when
+// connecting to Spanner.
+//
+// Database dialect can take 2 values: GoogleSQL or PostgreSQL and the same
+// correspond to regular Cloud Spanner database and PG Cloud Spanner database
+// respectively.
+//
+// If dbname is not specified, then HarbourBridge will autogenerate the same
+// and create a database with the same name.
+//
+// Example: -target-profile="instance=my-instance1,dbname=my-new-db1"
+// Example: -target-profile="instance=my-instance1,dbname=my-new-db1,dialect=PostgreSQL"
+//
+func NewTargetProfile(s string) (TargetProfile, error) {
+	params := make(map[string]string)
+	if len(s) > 0 {
```

Comment:
move this parsing logic into a common utility between source and destination, and just work on the output which should be a map

Reply:
Done.